compiler/rustc_error_codes/src/error_codes/E0802.md

```diff
@@ -0,0 +1,30 @@
+The `Async{Fn, FnMut, FnOnce}` traits are not yet `dyn`-compatible.
```

Comment:
Please respect the format: a short explanation of the error followed by a failing code example. Then all the explanation and at least one fixed code example.

Reply:
I've reordered this to move the examples up higher. PTAL.

If this still doesn't match what you had in mind, would you mind linking to an error example that does this properly? I used other error code files as a reference, but many are perhaps not idiomatic.

Reply:
I don't think this needs to be a new error code, frankly.